attestation-agent/kbs_protocol/src/client/mod.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#[cfg(feature = "background_check")]
+pub mod background_check;
+
+#[cfg(feature = "passport")]
+pub mod passport;
+
+use kbs_types::Tee;
+
+use crate::{keypair::TeeKeyPair, token_provider::Token};
+
+/// This Client is used to connect to the remote KBS.
+pub struct KbsClient<T> {
+    /// TEE Type
+    pub(crate) _tee: Option<Tee>,
```

Comment:
What does a (crate)-public `Option<Tee>` field intend to express?

An Option usually represents a value that may or may not exist in a struct. I suspect this is not case here and it's an implementation detail used it for lazy initialization.

If that's the case then we might want to make this field private, to prevent consuming code from messing with that in unintended ways. Maybe even use enum to document this intent:

```rust
enum ClientTee {
  Unitialized,
  Initializated(Tee),
}
```